### PR TITLE
add golang lint

### PR DESCRIPTION
**Pull request by changluyi**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
Examples of user facing changes:
- Features
- Bug fixes
- Docs
- Tests

### Which issue(s) this PR fixes:
Fixes #(issue-number)

### <samp>🤖 Generated by Copilot at ec95eea</samp>

Improved the linting workflow and configuration for the project. Added a `.github/workflows/.golangci.yml` file to customize the golangci-lint tool and modified the `.github/workflows/lint.yaml` file to use the new configuration and filter the relevant files.

### <samp>🤖 Generated by Copilot at ec95eea</samp>

> _`golangci.yml`_
> _Configures linters for code_
> _A winter project_

### <samp>🤖 Generated by Copilot at ec95eea</samp>

*  Add `.golangci.yml` file to configure the golangci-lint tool for the project ([link](https://github.com/kubeovn/kube-ovn/pull/3154/files?diff=unified&w=0#diff-3b1f34fa00cfbebaba8448b94d487590bbb6300049e8216161a88dd493034c9cR1-R31))
*  Restrict the linting action to run only when `.golangci.yml` file is changed ([link](https://github.com/kubeovn/kube-ovn/pull/3154/files?diff=unified&w=0#diff-4b122024a3a28ded65da76a2f1bface1f3a27328374438d1298d25585fe0603bR4-R5))
*  Use a list for the golangci-lint command line arguments and enable verbose output ([link](https://github.com/kubeovn/kube-ovn/pull/3154/files?diff=unified&w=0#diff-4b122024a3a28ded65da76a2f1bface1f3a27328374438d1298d25585fe0603bL41-R45))

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -828,6 +828,10 @@ lint:
 	@GOOS=linux go vet ./...
 	@GOOS=linux gosec -exclude=G204,G306,G404,G601,G301 -exclude-dir=test -exclude-dir=pkg/client ./...
 
+.PHONY: gofumpt
+gofumpt: gofumpt
+	find . -name "*.go" -execdir gofumpt -w -extra {} \;
```
Comment (oilbeater) [inline Makefile:833]: 这里用 -w 重写了，那 lint 的问题是不是就检查不出来了，但是 git 仓库里还没改

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (changluyi) [reply Makefile:833]: 是的 ，这个步骤不该放流水线里面。可以手动执行。

Comment (changluyi) [reply Makefile:833]: fixed

Comment (zhangzujian) [inline Makefile:833]:
```suggestion
	gofumpt -w -extra .
```

Comment (changluyi) [reply Makefile]: 1

**Review by oilbeater — APPROVED** [review]

**Pull request merged**